File: backend/springnatour/src/main/java/com/unina/springnatour/dto/chat/MessageDto.java
package com.unina.springnatour.dto.chat;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
public class MessageDto implements Serializable {
    @JsonProperty("message_id")
    private Long messageId;

    @JsonProperty("message_content")
    private String messageContent;

    @JsonProperty("sent_on")
    private LocalDateTime sentOn;

    @JsonProperty("sender_id")
    private Long senderId;

    @JsonProperty("recipient_id")
    private Long recipientId;

    @JsonProperty("chat_id")
    private Long chatId;
}

File: client/app/src/main/java/com/unina/natourkt/core/data/remote/dto/MessageDto.kt
package com.unina.natourkt.core.data.remote.dto

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MessageDto(
    @SerialName("message_id")
    @SerializedName("message_id")
    val messageId: Long,

    @SerialName("message_content")
    @SerializedName("message_content")
    val messageContent: String,

    @SerialName("sent_on")
    @SerializedName("sent_on")
    val sentOn: String,

    @SerialName("sender_id")
    @SerializedName("sender_id")
    val senderId: Long,

    @SerialName("recipient_id")
    @SerializedName("recipient_id")
    val recipientId: Long,

    @SerialName("chat_id")
    @SerializedName("chat_id")
    val chatId: Long,
)